### DISCUSSION TITLE
macOS: `NSWindowShouldDragOnGesture` ignored in macOS with `window-decoration = false`

### DISCUSSION THREAD

**Discussion by Uzaaft** [ghostty-org/ghostty#3053] [Issue Triage]
Running: `defaults write -g NSWindowShouldDragOnGesture -bool true` in the command line to enable this, and then restarting all windows, it appears that this doesnt work on ghostty. 

Putting it here since it didnt appear to be documented in the issues. 

Steps to reproduce:
1. Run: `defaults write -g NSWindowShouldDragOnGesture -bool true`
2. Restart ghostty(make sure to kill all windows) with the config: `window-decoration = false` enabled
3. Try and drag window with cmd+ctrl+click
This was only tested on the release variant of the application.

```
➤ /Applications/Ghostty.app/Contents/MacOS/ghostty --version
Ghostty 0.1.0-main+ae6c4c3f

Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.none
  - font engine: font.main.Backend.coretext
  - renderer   : renderer.Metal
  - libxev     : main.Backend.kqueue
 
```

**Comment by mitchellh** [comment]
You've got a good set of steps to reproduce here. I only converted to a discussion because I wasn't even aware of this default, so I wanted to do some research on it a bit more before converting to an issue someone might pick up (since our rules are if you make a PR to solve an issue I'll merge it).

**Reply by Uzaaft** [reply]
Fair enough. Would there be anything in the way of a PR from me if I figure out why this happens and come up with a fix?

**Reply by mitchellh** [reply]
No!

**Comment by justgook** [comment]
any updates on it?

**Comment by gempir** [comment]
Note, it's kind of impossible to drag the window with the mouse with window-deocrations disabled, so this feature is somewhat important for users who want to use their mouse sometimes.

How this feature works: https://mmazzarolo.com/blog/2022-04-16-drag-window-by-clicking-anywhere-on-macos/

**Comment by ShaiAvr** [comment]
I also noticed this bug today. I usually prefer to work without window decorations. On Linux, it's easy since I can use the super key to drag the window, but on my MacBook, Cmd+Ctrl+Drag isn't working without window decorations, which makes it impossible to drag a window with the mouse. Cmd+Ctrl+Drag does work when window decorations are enabled, though.

**Comment by bergman** [comment]
I am not sure what the difference is between `window-decoration = false` and `macos-titlebar-style = hidden` but moving windows with Cmd+Ctrl+Drag works for me with the latter (in combination with not setting `window-decoration` and having it fall back to the default `window-decoration = auto`).

The [documentation for `window-decoration`](https://ghostty.org/docs/config/reference#window-decoration) even suggests MacOS users to do that instead:
> macOS: To hide the titlebar without removing the native window borders or rounded corners, use `macos-titlebar-style = hidden` instead.



https://ghostty.org/docs/config/reference#macos-titlebar-style
